File: push.py
import subprocess
from datetime import datetime

def main():
    # 获取当前时间
    current_time = datetime.now()

    # 格式化时间戳
    timestamp = current_time.strftime("%Y-%m-%d %H:%M:%S")

    # 构造commit消息
    commit_message = "Site updated: " + timestamp

    # 执行git add命令
    try:
        subprocess.check_call(["git", "add", "-A"])
    except subprocess.CalledProcessError as e:
        print(e)
        exit(1)

    # 执行git commit命令
    try:
        subprocess.check_call(["git", "commit", "-m", commit_message])
    except subprocess.CalledProcessError as e:
        print(e)
        exit(1)

    # 执行git push命令
    try:
        subprocess.check_call(["git", "push"])
    except subprocess.CalledProcessError as e:
        print(e)
        exit(1)

    print("代码已成功推送到远程仓库！")

if __name__ == "__main__":
    main()

File: push.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include <sstream>
#include <cstdio>

std::string getCurrentTimestamp() {
    // Get the current time
    std::time_t currentTime = std::time(nullptr);

    // Format the timestamp
    char buffer[80];
    std::strftime(buffer, sizeof(buffer), "%Y-%m-%d %H:%M:%S", std::localtime(&currentTime));
    return std::string(buffer);
}

void executeCommand(const std::string& command) {
    // Execute the system command
    int result = std::system(command.c_str());
    if (result != 0) {
        std::cerr << "Command failed: " << command << std::endl;
        std::exit(EXIT_FAILURE);
    }
}

int main() {
    // Get the current formatted timestamp
    std::string timestamp = getCurrentTimestamp();

    // Construct the commit message
    std::stringstream commitMessage;
    commitMessage << "Site updated: " << timestamp;

    // Execute git add command
    executeCommand("git add -A");

    // Execute git commit command
    executeCommand("git commit -m \"" + commitMessage.str() + "\"");

    // Execute git push command
    executeCommand("git push");

    std::cout << "代码已成功推送到远程仓库！" << std::endl;

    return 0;
}